debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlValueConverters.java

```diff
@@ -253,10 +254,24 @@ protected Object convertEnumToString(String options, Column column, Field fieldD
             return data;
         }
         if (data instanceof Integer) {
-            // The binlog will contain an int with the 1-based index of the option in the enum value ...
-            int index = ((Integer) data).intValue() - 1; // 'options' is 0-based
-            if (index < options.length()) {
-                return options.substring(index, index + 1);
+
+            if (options != null) {
+                // The binlog will contain an int with the 1-based index of the option in the enum value ...
+                int index = ((Integer) data).intValue() - 1; // 'options' is 0-based
+                //playing with index is better, than splitting it into list each time. though have to handle error cases.
+                int startDeliminatorIndex = 0;
+                int nextDelimiatorIndex = options.indexOf(MySqlDdlParser.ENUM_AND_SET_DELIMINATOR);
+                while (index > 0 && nextDelimiatorIndex != -1) {
+                    startDeliminatorIndex = nextDelimiatorIndex + 1;
+                    nextDelimiatorIndex = options.indexOf(MySqlDdlParser.ENUM_AND_SET_DELIMINATOR, startDeliminatorIndex);
+                    index --;
+                }
+                //in case of last enum from the comma seperated string
+                if (index == 0 && nextDelimiatorIndex == -1) {
+                    return options.substring(startDeliminatorIndex);
+                } else if (nextDelimiatorIndex != -1) {
+                    return options.substring(startDeliminatorIndex,nextDelimiatorIndex);
+                }
```

Comment:
This is a method that is called with *every row change*, so if the options were passed in as a `List<String>` and this method becomes far simpler, it will also perform much better.